torch/autograd/function.py

```diff
@@ -296,12 +296,36 @@ def __init__(cls, name, bases, attrs):
 class _SingleLevelFunction(with_metaclass(FunctionMeta, _C._FunctionBase, FunctionCtx, _HookMixin)):  # type: ignore[misc]
     @staticmethod
     def forward(ctx: Any, *args: Any, **kwargs: Any) -> Any:
-        r"""Performs the operation.
+        r"""
+        This function is to be overridden by all subclasses. There are two ways
+        to define forward:
+
+        Usage 1 (Combined forward and ctx)::
+
+            @staticmethod
+            def forward(ctx: Any, *args: Any, **kwargs: Any) -> Any:
+                pass
+
+        - It must accept a context ctx as the first argument, followed by any
+          number of arguments (tensors or other types).
+        - See :ref:`combining-forward-context` for more details
+
+        Usage 2 (Separate forward and ctx)::
+
+            @staticmethod
+            def forward(*args: Any, **kwargs: Any) -> Any:
+                pass
+
+            @staticmethod
+            def setup_context(ctx: Any, inputs: Tuple[Any, ...], output: Any) -> None:
+                pass
 
-        This function is to be overridden by all subclasses.
 
-        It must accept a context ctx as the first argument, followed by any
-        number of arguments (tensors or other types).
+        - You must also define a setup_context staticmethod to handle setting up the
```

Comment:
nit: maybe add a point about: "forward no longer takes context ctx as the first argument."